pkg/services/apiserver/builder/helper.go

```diff
@@ -245,6 +245,18 @@ func SetupConfig(
 
 	serverConfig.EffectiveVersion = v
 
+	// set priority for aggregated discovery
+	for _, b := range builders {
+		gvs := GetGroupVersions(b)
+		if len(gvs) == 0 {
+			return fmt.Errorf("builder did not return any API group versions: %T", b)
+		}
+		pvs := scheme.PrioritizedVersionsForGroup(gvs[0].Group)
+		for i, gv := range pvs {
+			serverConfig.AggregatedDiscoveryGroupManager.SetGroupVersionPriority(metav1.GroupVersion(gv), 15000, len(pvs)-i)
```

Comment:
Question – are we OK with static priority value for groups? i.e. this means that all groups across all builders will have the same priority.

I think it's fine but I also think this could lead to undefined behaviour if we have multiple builders define the same group, but I guess that's better solved by throwing an error if multiple builders try to claim the same group.

Reply:
> are we OK with static priority value for groups? i.e. this means that all groups across all builders will have the same priority.

i wasn't too concerned about this breaking anything because the k8s code was setting them all to the same group and version priority by default, but i updated the behavior here to increment the group https://github.com/grafana/grafana/pull/101427/commits/0adf0e02860d6b20cfe3633bde38fc3ed60d743f to be safe

Reply:
Greath, thanks!